### ISSUE TITLE
deleting tools with bad names

### ISSUE THREAD

**Issue by keiranmraine** [ga4gh/dockstore#1201]
## Feature Request 

It would be nice if we could delete things from the unpublished list in the developer pages.

### Desired behaviour

Ability to remove an unpublished tool/workflow.  Something similar to the github deletion flow where you have to accurately enter the name of the thing you are trying to delete as confirmation.

I'm aware of the plans to split pub/unpub grouping but this is still valuable as it's relatively easy to create a tool with an error in the tool-name.

**Comment by denis-yuen** [comment]
I think the tricky bit here is that there is a desire by others to make workflows immutable. The reasoning for this is that once a tool/workflow has been run, it should never be possible to retroactively change it or delete it. 

That said, our current refresh approach and publishing/unpublishing model doesn't quite adhere to that, putting the onus on workflow developers.  I think deletion would exacerbate this. 

Thinking it through though, we could compromise with "Ability to remove an unpublished tool/workflow that has never been published before"

**Comment by keiranmraine** [comment]
If #1202 is implemented to allow testing of unpublished tools this would aid keeping things as "never previously published".

As you can't stop someone deleting a tag on quay.io you're always going to be reliant on the developer.

**Comment by denis-yuen** [comment]
See also #1995

**Comment by aofarrel** [comment]
Forking repos to add .dockstore.yml files has resulted in the addition of a lot of unpublished workflows in my Dockstore account. Maybe we can take into account whether or not the original source control repo has been deleted?

If I delete [aofarrel/fairyland](https://github.com/aofarrel/fairyland) from GitHub, its Dockstore entry would permanently break too. So, I should be allowed to delete it.

**Comment by denis-yuen** [comment]
Could be a nice to check to distinguish "toy" repos from "serious" repos with snapshots, zenodo exports, etc. 
i.e. if you managed to delete your repo outright, that's probably a sign it makes sense to delete from Dockstore regardless of reproducibility concerns

**Comment by coverbeck** [comment]
> Maybe we can take into account whether or not the source control repo has been deleted?

I don't think we can tell if it's been deleted or made private. But that's a real edge case, and maybe we should treat private repos like deleted repos (until/if we ever add private repo support).

**Comment by jason-cerrato** [comment]
I have a situation where I forked a repository with many workflows as a demonstration of how to set up a forked Github repository to work with Dockstore. I deleted the underlying Github repository after the demonstration but I still have that long list of unpublished workflows sitting in Dockstore. I would love to see this feature to delete unpublished workflows implemented!

**Comment by justinjj24** [comment]
Nice to have a workflow deletion/archive feature for number of cases instead of bookkeeping the details of unpublished workflows. [6533](https://discuss.dockstore.org/t/how-to-remove-duplicate-workflows/6533)

**Comment by aofarrel** [comment]
> Forking repos to add .dockstore.yml files has resulted in the addition of a lot of unpublished workflows in my Dockstore account.

This has gotten a little worse over time -- I'm the guy who wrote the .dockstore.yml templates, sometimes people erroneously credit me for their own workflows. If I fork their repos so I can open a PR to fix their .dockstore.yml, my fork will automatically trigger the creation of a new unpublished workflow in my Dockstore page, which is already a bit cluttered as-is. In other words, to remove myself from a workflow entry, I have to create a workflow entry.


> > Maybe we can take into account whether or not the source control repo has been deleted?
>
> I don't think we can tell if it's been deleted or made private. But that's a real edge case, and maybe we should treat private repos like deleted repos (until/if we ever add private repo support).

At the risk of suggesting something that might only benefit me and nobody else, this could be narrowed down to checking if it's a fork, and if that fork has been deleted. That's likely to prevent "false positives" where a repo goes private temporarily.

**Comment by denis-yuen** [comment]
Note GitHub itself suffers from a similar impersonation issue https://dev.to/martiliones/how-i-got-linus-torvalds-in-my-contributors-on-github-3k4g

Still a good idea to think through possible attacks via this route

On the original topic, we did recently add a form of deletion that may apply for many of the cases in this thread https://github.com/dockstore/dockstore/pull/5545 (never published forks, demos that were never shown to the public)

**Comment by emattei** [comment]
Hi, 
Is there any workaround that can be used?
On my side this has become almost unmanageable. Could we have at least an archived tab so I don't have workflows that have been dead for years popping up?

If I un-link adn re-link the organization, would that be a way to start fresh? 
Thanks

**Comment by denis-yuen** [comment]
> Is there any workaround that can be used?

Do you happen to know if https://github.com/dockstore/dockstore/pull/5545 would cover your use-case? (i.e. were these workflows ever published?)

> Could we have at least an archived tab so I don't have workflows that have been dead for years popping up?

In the case that the above doesn't work, we have had https://github.com/dockstore/dockstore/issues/2104 and https://github.com/dockstore/dockstore/issues/3199 though, so do vote/thumbs-up in those issues to show your support for those approaches as well.

**Comment by aofarrel** [comment]
#5545 is great, and be worth expanding this to include invalid unpublished workflows. A while back I forked a repo to add a .dockstore.yml and then deleted my fork. Seems that either I didn't add a valid .dockstore.yml in the first place, or Dockstore tried to update its copy and failed since the repo is deleted, but now I have a handful of unpublished workflows I cannot delete.

<img width="1263" alt="Screenshot 2023-08-30 at 11 54 34 AM" src="https://github.com/dockstore/dockstore/assets/27784612/3ec70d66-8b2a-414e-a5f8-2306822be4e3">

Separately, I'd like to request bulk-deletion. Doing it on the level of an entire repo would help prevent accidental deletions and would be a big help for those of us who have to fork large repos...
<img width="304" alt="Screenshot 2023-08-30 at 12 07 57 PM" src="https://github.com/dockstore/dockstore/assets/27784612/2538ea09-c485-4bfb-8fe5-1282d55c6c14">
